src/migrations/1747050057273-updateTutorialTemplate.ts

```
@@ -0,0 +1,69 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+const PLATFORM_SPACE_TUTORIALS = 'platform-space-tutorials'; // = enum at src/common/enums/template.default.type.ts
+
+export class UpdateTutorialTemplate1747050057273 implements MigrationInterface {
+  private findTemplateProfileId = async (
+    queryRunner: QueryRunner
+  ): Promise<string | null> => {
+    const templates: { id: string }[] = await queryRunner.query(`
+      SELECT profile.id FROM template
+        JOIN collaboration ON collaboration.id = template.collaborationId
+        JOIN callout ON callout.calloutsSetId = collaboration.calloutsSetId
+        JOIN callout_framing ON callout_framing.id = callout.framingId
+        JOIN profile ON profile.id = callout_framing.profileId
+      WHERE template.id IN (
+        SELECT templateId FROM template_default WHERE templatesManagerId IN (
+          SELECT templatesManagerId FROM platform
+        ) AND template_default.type = '${PLATFORM_SPACE_TUTORIALS}'
+      )
+      AND profile.description LIKE 'Done with the tutorials and ready to%';
+    `);
+    if (templates.length === 0) {
+      console.warn('No templates to update');
+      return null;
+    } else if (templates.length > 1) {
+      throw new Error(
+        `Found ${templates.length} templates to update, expected only one.`
+      );
+    } else {
+      return templates[0].id;
+    }
+  };
+
+  private updateDescription = async (
+    queryRunner: QueryRunner,
+    profileId: string,
+    newDescription: string
+  ) => {
+    await queryRunner.query(
+      `
+      UPDATE profile SET description = ? WHERE id = ?;
+    `,
+      [newDescription, profileId]
+    );
+  };
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    // Updates the text of certain callout in the platform template to create a space with tutorials
+    const profileId = await this.findTemplateProfileId(queryRunner);
+    if (profileId) {
```

if null - throw, to avoid silently skipping

---

throwing in the previous function now